libraries/abstractions/common_io/test/iot_test_common_io.c

```diff
@@ -88,4 +88,116 @@ TEST_GROUP_RUNNER( Common_IO )
             RUN_TEST_GROUP( TEST_IOT_SPI );
         }
     #endif
+
+    #ifdef IOT_TEST_COMMON_IO_WATCHDOG_SUPPORTED
+        for( i = 0; i < IOT_TEST_COMMON_IO_WATCHDOG_SUPPORTED; i++ )
+        {
+            SET_TEST_IOT_WATCHDOG_CONFIG( i );
+            RUN_TEST_GROUP( TEST_IOT_WATCHDOG );
+        }
+    #endif
+
+    #ifdef IOT_TEST_COMMON_IO_RTC_SUPPORTED
+        for( i = 0; i < IOT_TEST_COMMON_IO_RTC_SUPPORTED; i++ )
+        {
+            SET_TEST_IOT_RTC_CONFIG( i );
+            RUN_TEST_GROUP( TEST_IOT_RTC );
+        }
+    #endif
+
+    #ifdef IOT_TEST_COMMON_IO_RESET_SUPPORTED
+        for( i = 0; i < IOT_TEST_COMMON_IO_RESET_SUPPORTED; i++ )
+        {
+            SET_TEST_IOT_RESET_CONFIG( i );
+            RUN_TEST_GROUP( TEST_IOT_RESET );
+        }
+    #endif
+
+    #ifdef IOT_TEST_COMMON_IO_PERFCOUNTER_SUPPORTED
+        for( i = 0; i < IOT_TEST_COMMON_IO_PERFCOUNTER_SUPPORTED; i++ )
+        {
+            SET_TEST_IOT_PERFCOUNTER_CONFIG( i );
+            RUN_TEST_GROUP( TEST_IOT_PERFCOUNTER );
+        }
+    #endif
+
+    #ifdef IOT_TEST_COMMON_IO_FLASH_SUPPORTED
+        for( i = 0; i < IOT_TEST_COMMON_IO_FLASH_SUPPORTED; i++ )
+        {
+            SET_TEST_IOT_FLASH_CONFIG( i );
+            RUN_TEST_GROUP( TEST_IOT_FLASH );
+        }
+    #endif
+
+    #ifdef IOT_TEST_COMMON_IO_TIMER_SUPPORTED
```

Comment:
Should the Timer"test case be located before the Flash tests? Reason for the question is that the Flash tests depend on the Timer APIs which means it make sense (for me) to run these tests prior to the Flash ones (assuming you know about this dependency it might help potential debugging).

Reply:
Good catch, I'll move this